avm/ptn/azd/ml-project/main.bicep

```diff
@@ -0,0 +1,171 @@
+
+@description('Required. The resource ID of the AI Studio Hub Resource where this project should be created.')
+param hubResourceId string
+
+@description('Required. The name of the key vault.')
+param keyVaultName string
+
+@description('Optional. The managed identity definition for the machine learning resource. At least one identity type is required.')
+param projectManagedIdentities managedIdentitiesType = {
+  systemAssigned: true
+}
+
+@description('Required. The name of the user assigned identity.')
+param userAssignedtName string
+
+@description('Optional. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleDefinitionIdOrName array = [
+  'f6c7c914-8db3-469d-8ca1-694a8f32e121'
+  'ea01e6af-a1c1-4350-9563-ad00f8c72ec5'
+]
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-mlproject.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module project 'br/public:avm/res/machine-learning-services/workspace:0.7.0' = {
+  name: '${uniqueString(deployment().name, location)}-project'
+  params: {
+    name: name
+    tags: tags
+    sku: projectSku
+    kind: projectKind
+    managedIdentities: projectManagedIdentities
+    hbiWorkspace: hbiWorkspace
+    publicNetworkAccess: publicNetworkAccess
+    hubResourceId: hubResourceId
+  }
+}
+
+module keyVaultAccess 'br/public:avm/res/key-vault/vault:0.9.0' = {
+  name: '${uniqueString(deployment().name, location)}-keyvault'
+  params: {
+    name: keyVault.name
+    accessPolicies: [
+      {
+        objectId: project.outputs.systemAssignedMIPrincipalId
+        permissions: { secrets: [ 'get', 'list' ] }
+      }
+    ]
+  }
+}
+
+module mlServiceRoleAssigned 'br/public:avm/res/managed-identity/user-assigned-identity:0.4.0' = {
+  name: '${uniqueString(deployment().name, location)}-roleassignment'
+  params: {
+    name: userAssignedtName
+    location: location
+    roleAssignments: [
+      for id in roleDefinitionIdOrName: {
+        principalId: project.outputs.systemAssignedMIPrincipalId
+        roleDefinitionIdOrName: id
+        principalType: 'ServicePrincipal'
+      }
+    ]
+  }
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
```

Comment:
Probably cosmetic, but imho it would help the readability of the module if resource and module declaration followed the deployment order